fuse/plugins/truth_information/peak_truth.py

```diff
@@ -0,0 +1,160 @@
+             ('average_z_of_contributing_clusters', np.float32),
+             ('average_x_obs_of_contributing_clusters', np.float32),
+             ('average_y_obs_of_contributing_clusters', np.float32),
+             ('average_z_obs_of_contributing_clusters', np.float32),
+            ]
+    dtype = dtype + strax.time_fields
+
+    gain_model_mc = straxen.URLConfig(
+        default="cmt://to_pe_model?version=ONLINE&run_id=plugin.run_id",
+        infer_type=False,
+        help='PMT gain model',
+    )
+
+    pmt_circuit_load_resistor = straxen.URLConfig(
+        default = "take://resource://"
+                  "SIMULATION_CONFIG_FILE.json?&fmt=json"
+                  "&take=pmt_circuit_load_resistor",
+        type=(int, float),
+        cache=True,
+        help='PMT circuit load resistor', 
+    )
+
+    digitizer_bits = straxen.URLConfig(
+        default = "take://resource://"
+                  "SIMULATION_CONFIG_FILE.json?&fmt=json"
+                  "&take=digitizer_bits",
+        type=(int, float),
+        cache=True,
+        help='Number of bits of the digitizer boards',
+    )
+
+    digitizer_voltage_range = straxen.URLConfig(
+        default = "take://resource://"
+                  "SIMULATION_CONFIG_FILE.json?&fmt=json"
+                  "&take=digitizer_voltage_range",
+        type=(int, float),
+        cache=True,
+        help='Voltage range of the digitizer boards',
+    )
+
+    max_drift_length = straxen.URLConfig(
+        default=straxen.tpc_z,
+        type=(int, float),
+        help="Total length of the TPC from the bottom of gate to the top of cathode wires [cm]",
+    )
+
+    drift_velocity_liquid = straxen.URLConfig(
+        default = "take://resource://"
+                  "SIMULATION_CONFIG_FILE.json?&fmt=json"
+                  "&take=drift_velocity_liquid",
+        type=(int, float),
+        cache=True,
+        help='Drift velocity of electrons in the liquid xenon',
+    )
+
+    def setup(self):
+        super().setup()
+
+        self.gains = pmt_gains(self.gain_model_mc,
+                               digitizer_voltage_range=self.digitizer_voltage_range,
+                               digitizer_bits=self.digitizer_bits,
+                               pmt_circuit_load_resistor=self.pmt_circuit_load_resistor
+                               )
+
+    def get_window_size(self):
+
+        drift_time_max = int(self.max_drift_length / self.drift_velocity_liquid)
+        
+        return drift_time_max * 20
+
+    def compute(self,interactions_in_roi, propagated_photons, peaks):
+
+        n_peaks = len(peaks)
+
+        result = np.zeros(n_peaks, dtype=self.dtype)
+        result['time'] = peaks['time']
+        result['endtime'] = peaks['endtime']
+
+        photons_in_peaks = strax.split_by_containment(propagated_photons, peaks)
+
+        for i in range(n_peaks):
+
+            s1_photon_cut = photons_in_peaks[i]["photon_type"] == 1
```

Comment:
It might be possible to loop through S1, S2, AP, so that we do not make errors in these similar copy-pasted codes.

Reply:
Makes sense! I tried to implement it, but it is not a 100% clean solution as I need to have data for S1 and S2 after the loop.